utl/dump_options/main.cc

```diff
@@ -284,6 +284,9 @@ struct options_exporter : options_i
       option_obj.AddMember("short_name", short_name_value, allocator);
       option_obj.AddMember("keep", option->m_keep, allocator);
       option_obj.AddMember("necessary", option->m_necessary, allocator);
+      rapidjson::Value one_of_value;
+      one_of_value.SetString(option->m_one_of_str, allocator);
+      option_obj.AddMember("one_of", one_of_value, allocator);
```

Comment:
I feel like this should only be added if the option is in fact a one_of option? Additionally, this should be a list of possible values and not just a formatted string

Reply:
Would it be sufficient to use a list of possible values, each as a string? (as opposed to string or int). Since untyped base_option's are passed in here, it wouldn't be possible to maintain the types for one_of unless we typed base_option, or used some caching scheme to decipher the type here. Also short_name is included regardless of being set so I'll also change that to only include if it's set.

Reply:
The base option has a type id you can use to dynamic cast the option. That's what we're doing for the other things.

Sounds like a good idea to fix short option. Maybe we should use `null` to indicate a missing optional value?

Reply:
I just used the empty string to detect if a short name is not present, since this is the default value if nothing is indicated.